Xamarin.Forms.Core/Shell/Shell.cs

```diff
@@ -324,9 +322,7 @@ void IShellController.UpdateCurrentState(ShellNavigationSource source)
 			OnNavigated(new ShellNavigatedEventArgs(oldState, CurrentState, source));
 		}
 
-#endregion IShellController
-
-#region URI Navigation
+		public static Shell CurrentShell => Application.Current?.MainPage as Shell ?? throw new InvalidOperationException("Application.Current.MainPage is not a Shell");
```

Comment:
is it necessary to throw an exception here?

Reply:
I'm wondering the same thing. Do we have any mechanism for a component to determine if it's running in a Shell app (as opposed to a classic app)? If not, does it make sense to allow a check for `CurrentShell == null`?

Reply:
go one
```suggestion
		public static Shell CurrentShell => Application.Current?.MainPage as Shell;
```